utils/ixdtf/src/parsers/time_zone.rs

```diff
@@ -0,0 +1,239 @@
+    /// TimeZone Data
+    pub(crate) tz: TimeZone<'a>,
+}
+
+// ==== Time Zone Annotation Parsing ====
+
+pub(crate) fn parse_ambiguous_tz_annotation<'a>(
+    cursor: &mut Cursor<'a>,
+) -> ParserResult<Option<TimeZoneAnnotation<'a>>> {
+    // Peek position + 1 to check for critical flag.
+    let mut current_peek = 1;
+    let critical = cursor
+        .peek_n_char(current_peek)
+        .map(is_critical_flag)
+        .ok_or(ParserError::abrupt_end("AmbiguousAnnotation"))?;
+
+    // Advance cursor if critical flag present.
+    if critical {
+        current_peek += 1;
+    }
+
+    let leading_char = cursor
+        .peek_n_char(current_peek)
+        .ok_or(ParserError::abrupt_end("AmbiguousAnnotation"))?;
+
+    if is_tz_leading_char(leading_char) || is_sign(leading_char) {
+        // Ambigious start values when lowercase alpha that is shared between `TzLeadingChar` and `KeyLeadingChar`.
+        if is_a_key_leading_char(leading_char) {
+            let mut peek_pos = current_peek + 1;
+            while let Some(ch) = cursor.peek_n_char(peek_pos) {
+                if is_tz_name_separator(ch) || (is_tz_char(ch) && !is_a_key_char(ch)) {
+                    let tz = parse_tz_annotation(cursor)?;
+                    return Ok(Some(tz));
+                } else if is_annotation_key_value_separator(ch)
+                    || (is_a_key_char(ch) && !is_tz_char(ch))
+                {
+                    return Ok(None);
+                } else if is_annotation_close(ch) {
+                    return Err(ParserError::InvalidAnnotation);
+                }
+
+                peek_pos += 1;
+            }
+            return Err(ParserError::abrupt_end("AmbiguousAnnotation"));
+        }
+        let tz = parse_tz_annotation(cursor)?;
+        return Ok(Some(tz));
+    }
+
+    if is_a_key_leading_char(leading_char) {
+        return Ok(None);
+    };
+
+    Err(ParserError::AnnotationChar)
+}
+
+fn parse_tz_annotation<'a>(cursor: &mut Cursor<'a>) -> ParserResult<TimeZoneAnnotation<'a>> {
+    assert_syntax!(
+        is_annotation_open(cursor.next_or(ParserError::AnnotationOpen)?),
+        AnnotationOpen
+    );
+
+    let critical = cursor.check_or(false, is_critical_flag);
+    cursor.advance_if(critical);
+
+    let tz = parse_time_zone(cursor)?;
+
+    assert_syntax!(
+        is_annotation_close(cursor.next_or(ParserError::AnnotationClose)?),
+        AnnotationClose
+    );
+
+    Ok(TimeZoneAnnotation { critical, tz })
+}
+
+/// Parses the [`TimeZoneIdentifier`][tz] node.
+///
+/// [tz]: https://tc39.es/proposal-temporal/#prod-TimeZoneIdentifier
+pub(crate) fn parse_time_zone<'a>(cursor: &mut Cursor<'a>) -> ParserResult<TimeZone<'a>> {
+    let is_iana = cursor
+        .check(is_tz_leading_char)
+        .ok_or(ParserError::abrupt_end("TimeZoneAnnotation"))?;
+    let is_offset = cursor.check_or(false, is_sign);
+
+    if is_iana {
+        return parse_tz_iana_name(cursor);
+    } else if is_offset {
```

Comment:
Question: is this code path for something like `[-08:00]`? Does the IXDTF syntax allow that?

Reply:
I'm fairly certain it does in both Temporal and Sedate.

Sedate lists a time zone annotation as below

```txt
Sedate:
time-zone         = "[" critical-flag
                        time-zone-name / time-numoffset "]"

RFC3339 5.6:
time-numoffset  = ("+" / "-") time-hour ":" time-minute
```